### PR TITLE
Add changes related to scope management

### PR DESCRIPTION
**Pull request by dushaniw**
## Purpose
Update docs related to shared scopes and scope management.
Add depecrated note for API scopes feature.

### CODE REVIEW

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 1]
```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
...
+     title="Resource Scopes API Definition" width="50%" /></a>
+
+[![Scopes Security Definition API Definition]({{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/scopes-api-definition.png)]({{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/scopes-api-definition.png)
+
 
 ## Obtaining Tokens with Scopes
 
-When a scope has been assigned to an API resource, the access is getting restricted to the users who has the role/roles specified in the scope. Hence, the API consumers has to specifically request for the relevant scope when generating an access token to invoke the API resource. Please follow below steps to obtain an access token specifying the requested scopes.
+When a scope has been assigned to an API resource, the access is getting restricted to the users who has the role/roles specified in the scope. Hence, the API consumers has to specifically request for the relevant scope when generating an access token to invoke the API resource. Follow below steps to obtain an access token specifying the requested scopes.
```
Comment (shaniR) [inline en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:185]: Shall we rephrase the above to : When a scope is attached to an API resource, access to it gets restricted based on the role(s) that is specified in the scope. In order to invoke the API resource, the API consumer has to generate an access token bound to the scope that is attached to the API resource. Follow below steps to obtain an access token specifying the requested scopes.

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 2]
```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
...
-3. Generate production keys for the application.
+3. Generate PRODUCTION or SANDBOX keys for the application.
 
-    <a href="../../../../../assets/img/learn/generate-app-keys.png" ><img src="../../../../../assets/img/learn/generate-app-keys.png" alt="Generate App Keys" 
-         title="Generate App Keys" width="60%" /></a>
+     <a href="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/generate-keys.png" ><img src="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/generate-keys.png" alt="Generate App Keys" 
+          title="Generate App Keys" width="60%" /></a>
 
-4. Click on **GENERATE ACCESS TOKEN** button. Then select the scope from the prompted window and generate the access token.
+4. Click on **GENERATE ACCESS TOKEN** button. Then select the scope from dropdown in the prompted window and click **GENERATE** button
```
Comment (shaniR) [inline en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:199]: Click **GENERATE ACCESS TOKEN**

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 3]
```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
...
+3. Generate PRODUCTION or SANDBOX keys for the application.
 
-    <a href="../../../../../assets/img/learn/generate-app-keys.png" ><img src="../../../../../assets/img/learn/generate-app-keys.png" alt="Generate App Keys" 
-         title="Generate App Keys" width="60%" /></a>
+     <a href="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/generate-keys.png" ><img src="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/generate-keys.png" alt="Generate App Keys" 
+          title="Generate App Keys" width="60%" /></a>
 
-4. Click on **GENERATE ACCESS TOKEN** button. Then select the scope from the prompted window and generate the access token.
+4. Click on **GENERATE ACCESS TOKEN** button. Then select the scope from dropdown in the prompted window and click **GENERATE** button 
+to generate the access token.
```
Comment (shaniR) [inline en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:200]: unnecessary spaces before this line

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 4]
```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
...
 
-    <a href="../../../../../assets/img/learn/generate-access-token-with-scopes.png" ><img src="../../../../../assets/img/learn/generate-access-token-with-scopes.png" alt="Generate App Keys" 
-         title="Generate App Keys" width="50%" /></a>
+     <a href="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/generate-keys.png" ><img src="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/token-scopes.png" alt="Select Token Scopes" 
+          title="Select Token Scopes" width="60%" /></a>
 
-5. If the user has the roles specified in the scope, the access token will be issued with the requested scope. Unless, only the default scopes will be returned with the access token.
+5. If the user has the roles specified in the scope, the access token will be issued with the requested scope. Unless,
+only the default scopes will be returned with the access token.
```
Comment (shaniR) [inline en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:206]: unncessary space in this line

Comment (shaniR) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:206]: Unless -> Otherwise

Comment (shaniR) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:206]: What are the default scopes? how to define them? where are they defined?

**Comment by Mariangela** [comment]
@dushaniw , can you please address the review comments.

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 5]
```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
...
+
+### Creating a Shared Scope
+
+The shared scope need to be created before API creation/update time. Follow below steps to create a new shared scope.
+
+1. Login to API Publisher (https://localhost:9443/publisher). 
+
+2. Navigate to **Scopes** from the top menu and click on **Create a new scope** button.
+
+     [![Start Creating a Scope]({{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/create-shared-scope.png)]({{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/create-shared-scope.png)
```
[the remark this answers is not in the document]
Comment (Mariangela) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:27]: @shaniR , this is correct. It is done so that when a user clicks on the image the image will appear zoomed.

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 6]
```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
...
+
+3. In **Create New Scope** wizard, you will be prompted to enter a scope name, scope display name and optionally,
```
[the remark this answers is not in the document]
Comment (dushaniw) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:29]: the name of the wizard is "Create New Scope"

[the remark this answers is not in the document]
Comment (dushaniw) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:29]: image

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 7]
```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
 
 For example, there can be requirements such as restricting the access to a given API resource to admin users only, while other resources of the same API access should be allowed to consumers with less privileges. Let's see how this kind of role based access control can be managed with the use of OAuth2 scopes.
 
+## Shared Scopes
```
[the remark this answers is not in the document]
Comment (dushaniw) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:7]: Shared OAuth Scopes

need the differentiation from API-M side, as there are local oauth scopes as well

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 8]
```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
...
          title="Apply Scope" width="70%" /></a>
 
-3. Publish the API.
+3. Click on **Save** button to save changes.         
+
+4. Publish the API.
+
+## Applying Multiple Scopes Per Resource
+
+From API-M Publisher Portal, you can attach multiple scopes to each API resource including combinations of both
```
[the remark this answers is not in the document]
Comment (dushaniw) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:162]: but we still provide the UI support

[the remark this answers is not in the document]
Comment (dushaniw) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:162]: will remove this.

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md (hunk 1 above)
Comment (dushaniw) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:185]: +1

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md (hunk 4 above)
Comment (dushaniw) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:206]: its the default scope. in client credentials pov, this will also include "am_application_scope". Will add a note.  Btw fyi this is also in the 3.1 and 3.0 docs well.

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 9]
```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
 
 For example, there can be requirements such as restricting the access to a given API resource to admin users only, while other resources of the same API access should be allowed to consumers with less privileges. Let's see how this kind of role based access control can be managed with the use of OAuth2 scopes.
 
+## Shared Scopes
 
-## Creating a Scope
+In WSO2 API-M, you can create OAuth scopes before the API creation time and share them across multiple APIs of the same
```
[the remark this answers is not in the document]
Comment (dushaniw) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md]: +1

**Comment by dushaniw** [comment]
> @dushaniw , can you please address the review comments.

done

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (shaniR) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:26]: noted.

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 10]
```diff
@@ -4,95 +4,216 @@ Scopes enable fine-grained access control to API resources based on user roles.
...
-    <a href="../../../../../assets/img/learn/create-a-new-scope.png" ><img src="../../../../../assets/img/learn/create-a-new-scope.png" alt="Create a Scope" 
+    <a href="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/api-scopes/create-a-new-scope.png" >
+    <img src="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/api-scopes/create-a-new-scope.png" alt="Create a Scope" 
          title="Create a Scope" width="70%" /></a>
 
-## Applying a Scope to an API Resource
+### Applying a Local Scope to a Resource
 
-A scopes has be applied to a resource, in order to restrict the access to a user group/groups. Please follow below steps to apply a scope to a resource.
+A Local scope has be applied to a resource, in order to restrict the access to a user group/groups. A Local Scope can only be
```
Comment (shaniR) [inline en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:140]: Shall we rephrase it to : "A local scope, when applied to a resource, will restrict access for a user group(s)."

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 11]
```diff
@@ -4,95 +4,216 @@ Scopes enable fine-grained access control to API resources based on user roles.
...
 
-1. Go to API Resource section and click on the resource you would like to apply the scope.
+1. Go to API **Resources** section and click on the resource you would like to apply the scope.
 
-    <a href="../../../../../assets/img/learn/select-resource.png" ><img src="../../../../../assets/img/learn/select-resource.png" alt="Select Resource" 
+    <a href="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/select-resource.png" ><img src="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/select-resource.png" alt="Select Resource" 
          title="Select Resource" width="70%" /></a>
 
-2. Select the created scope from the drop down and click on **Save** button to save changes.
+2. Select the created Local Scope from the dropdown. The created local scope will be listed under **API Scopes** in the
```
Comment (shaniR) [inline en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:148]: Local Scope - local scope

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md [hunk 12]
```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
...
-2. Select the created scope from the drop down and click on **Save** button to save changes.
+2. Select the created API Scope from the dropdown. The created shared scope will be listed under **API Scopes** in the 
+dropdown.
 
-    <a href="../../../../../assets/img/learn/apply-scope.png" ><img src="../../../../../assets/img/learn/apply-scope.png" alt="Apply Scope" 
+    <a href="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/api-scopes/apply-scope.png" ><img src="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/api-scopes/apply-scope.png" alt="Apply Scope" 
          title="Apply Scope" width="70%" /></a>
 
-3. Publish the API.
+3. Click on **Save** button to save changes.
```
[the remark this answers is not in the document]
Comment (shaniR) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md:154]: Shall we add a note mentioning the limitation that the local scope gets deleted when a global scope is created with the same name?

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md (hunk 11 above)
Comment (dushaniw) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md]: done

File: en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md (hunk 10 above)
Comment (dushaniw) [reply en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md]: done
